### PR TITLE
✨amp-consent: consent-ui Allow passing in the reason for prompt in the iframe name attribute

### LINKED ISSUE #21626: amp-consent: Specify in CMP iframe `name` attribute reason for loading
Would like to know for example, if the CMP Iframe was loaded by `forcePromptOnNext` or `on="tap:consent.prompt"` or if it is the default bahavior. This should be passed in from the iframe name JSON.

### PR DESCRIPTION
**Pull request by torch2424**
closes #21626

As the title states! Please take a look at the second example, to make sure this is the expected behavior.

# Example

**Prompted by clicking the manage button on the postPromptUI**

Screen Shot 2019-03-29 at 3 42 35 PM

**Prompted by there being no consent string (No postPrompt because there is no identifiable trigger right?)**

Screen Shot 2019-03-29 at 3 42 56 PM

### CODE REVIEW

File: extensions/amp-consent/0.1/amp-consent.js [hunk 1]
```diff
@@ -54,6 +54,14 @@ export const ACTION_TYPE = {
   DISMISS: 'dismiss',
 };
 
+/**
+ * @enum {string}
+ */
+export const PROMPT_TRIGGER = {
+  ACTION: 'action',
+  FORCE_PROMPT_ON_NEXT: 'force-prompt-on-next',
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/amp-consent.js]: Thank you for making the change!! 'force-prompt-on-next' is no difference than the initial prompt. I don't think we need it. What about we use a boolean here instead?

Comment (torch2424) [reply extensions/amp-consent/0.1/amp-consent.js]: @zhouyx Can we discuss this offline tomorrow? I'm a little confused why we would use a boolean 🤔 . That being said, I wouldn't be opposed to it if it makes sense. 😄

File: extensions/amp-consent/0.1/consent-ui.js [hunk 2]
```diff
@@ -438,6 +440,9 @@ export class ConsentUI {
     toggle(dev().assertElement(this.ui_), false);
 
     const iframePromise = this.getClientInfoPromise_().then(clientInfo => {
+      if (opt_promptTrigger) {
+        clientInfo['promptTrigger'] = opt_promptTrigger;
```
Comment (torch2424) [inline extensions/amp-consent/0.1/consent-ui.js]: `opt_promptTrigger` should be a boolean internally,

the line would be come:

clientInfo['promptTrigger'] = opt_promptTrigger ? 'action' : 'load';

`action` - Caused by an action, like a button click.

`load` - Prompt was triggered on page load (force prompt, unknown consent state) 

This is in response to @zhouyx 's comment. Thanks for helping in the bikeshedding @calebcordry 😄

**Comment by torch2424** [comment]
@zhouyx This is good to go! 😄

File: extensions/amp-consent/0.1/consent-ui.js [hunk 3]
```diff
@@ -438,6 +440,7 @@ export class ConsentUI {
     toggle(dev().assertElement(this.ui_), false);
 
     const iframePromise = this.getClientInfoPromise_().then(clientInfo => {
+      clientInfo['promptTrigger'] = isActionPromptTrigger ? 'action' : 'load';
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-ui.js]: Nit: Can we move the `promptTrigger'` initiation to within the `getClientInfoPromise_` function? I think it makes more sense to complete initiation within one function. Thanks.

**Pull request merged**